torch/export/exported_program.py

```diff
@@ -211,10 +201,18 @@ def _override_composite_implicit_decomp(cia_ops_to_callable, safe=True):
             )
 
         # For fake tensor prop, we do want to register meta kernel directly
-        if torch._C.DispatchKey.Meta not in op_overload.py_kernels:
-            op_overload.py_impl(torch._C.DispatchKey.Meta)(
-                functools.partial(_register_cia_to_meta, kernel=op_overload)
-            )
+        if (
+            torch._C.DispatchKey.Meta not in op_overload.py_kernels
+        ):  
+            # [NOTE] Registering old CIA to Meta kernel
+            # We always register original CIA behavior to the Meta kernel
+            # The reason is when we are fake tensor prop-ing, we end up calling 
+            # an operator on Meta backend, which in python dispatcher, will resolve 
+            # into CIA key. (see resolve_key in torch/_ops.py) As a result, this CIA
+            # now will call into the custom user defined CIA which can cause a problem.
+            # Since we only run meta impl for shape prop, we should always use the default
+            # behavior. 
+            op_overload.py_impl(torch._C.DispatchKey.Meta)(orig_cia_callable)
```

Comment:
Can you add some OSS tests? In particular, the one you had that involved:

(1) the model is using Linear()
(2) you export the model with `aten.linear()` not decomposed
(3) dynamic shapes are enabled

Reply:
Added!